### DISCUSSION TITLE
java-agent 8.16.0, The 'agent.ignore_suffix' parameter in the agent.config does not seem to be taking effect

### DISCUSSION THREAD

**Discussion by hzhswyz** [apache/skywalking#11136] [Q&A]
My service has an api with a path of /api/getStock. Due to the batch redis operations performed (hundreds of times) in this api, the number of spans in a single segment exceeded 350, resulting in the following error.
<img width="1103" alt="image" src="https://github.com/apache/skywalking/assets/44084348/b17627d4-dfd2-4a93-9c1e-849a67fd4a77">

My api path
![image](https://github.com/apache/skywalking/assets/44084348/ed9c0e0e-193e-4de7-9c23-99cce8c5a443)

Therefore, I want to ignore the trace of this request. I see that the agent.config has a parameter 'agent.ignore_suffix' can ignore urls with specific suffixes, so I added the suffix ‘getStock’, but it doesn't seem to take effect. What is the reason?
<img width="1365" alt="image" src="https://github.com/apache/skywalking/assets/44084348/5607b71a-5d40-49cf-978e-a40835aff750">

I can still see the trace of this path.
![image](https://github.com/apache/skywalking/assets/44084348/09569054-b0d4-4f78-a811-ad239a1f3aca)

**Comment by wu-sheng** [comment]
You need to debug on your own first. We would NOT be able to guess what happens there.

**Reply by hzhswyz** [reply]
According to the source code, If want url to be ignored, operationName must contain "." , which means our path needs to have "." . For restful style urls, it is unlikely to contain '".". I think that should be specified in the 'agent.ignore_suffix' description.
![image](https://github.com/apache/skywalking/assets/44084348/862c9819-4fc1-4532-9807-c8c961882a0a)

**Reply by hzhswyz** [reply]
I found an optional plugin apm-trace-ignore-plugin , and after testing, it can ignore specified urls. It took me a lot of time debugging to make the feature work as expected because the documentation did not mention that the URL pattern must contain a request method prefix.
<img width="1071" alt="image" src="https://github.com/apache/skywalking/assets/44084348/f39290bd-63fe-42d3-8817-bff273c6542f">
Actually, it should be "POST:/api/getStock". Another issue is that if other services call this specified ignored URL through RPC, the agent will still continue to report traces.

**Reply by wu-sheng** [reply]
> it is unlikely to contain '".". I think that should be specified in the 'agent.ignore_suffix' description.

This is a totally what suffix means. It is never designed for your use cases. All existing configurations are about not tracing resource requests.

> Actually, it should be "POST:/api/getStock"

The HTTP is, but others are not. Your case is not the only user case. There could be gRPC, Dubbo, timer, internal timer, MQ consumer, etc.

> Another issue is that if other services call this specified ignored URL through RPC, the agent will still continue to report traces.

Yes. Same as before, you thought your current use case is the only case. If the trace doesn't continue, you will complain trace breaks unexpectedly. If this is being called by others through distributed chain, you should disable the whole chain.

**Reply by wu-sheng** [reply]
Generally, open source has provided you with enough information, and the documentation could be polished, but you will see that it is not easy.
